Examples/HackerNewsClient/HackerNewsClient/Providers/PostsProvider.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  PostsProvider.swift
+//  HackerNewsClient
+//
+//  Created by Francesco Di Lorenzo on 15/11/2016.
+//  Copyright © 2016 BendingSpoons. All rights reserved.
+//
+
+import Katana
+
+extension PostsProvider {
+    static let hnFrontPageURL = URL(string: "https://hn.algolia.com/api/v1/search?tags=front_page")!
+}
+
+struct PostsProvider: SideEffectDependencyContainer {
+    var posts: [Post]?
+
+    init(state: State, dispatch: @escaping StoreDispatch) {}
+    
+    public func fetchPosts(completion: @escaping ([Post]?, String?) -> ()) {
+        
+        let task = URLSession.shared.dataTask(with: PostsProvider.hnFrontPageURL) { (data, response, error) in
+            
+            if let error = error {
+                completion(nil, error.localizedDescription)
+                return
+            }
+            
+            guard let data = data,
```

Comment:
I think you can refactor this code (anche the one that creates the posts) using `flatMap`

For instance:
```swift
let hits = data
   .flatMap { try? JSONSerialization.jsonObject(with: $0, options: []) }
   .flatMap { $0 as? [String: Any] }
```